src/spikeinterface/extractors/mdaextractors.py

```diff
@@ -216,10 +216,14 @@ def write_sorting(sorting, save_path, write_primary_channels=False):
         times_list = []
         labels_list = []
         primary_channels_list = []
-        for unit_id in unit_ids:
+        for unit_id_i, unit_id in enumerate(unit_ids):
             times = sorting.get_unit_spike_train(unit_id=unit_id)
             times_list.append(times)
-            labels_list.append(np.ones(times.shape) * unit_id)
+            # unit id may not be numeric
+            if unit_id.dtype.kind in "biufc":
+                labels_list.append(np.ones(times.shape) * unit_id)
```

Comment:
unit_id is int or str only. float are not allowed

Reply:
I seem to recall I've had this (float unit ids). How to deal with that, an assert here or just allow it?

Reply:
No by construction they can only be int or str. Otherwise an error is raised in base